lambda/lambda.ml

```diff
@@ -892,5 +892,8 @@ let function_is_curried func =
   | Curried -> true
   | Tupled -> false
 
+let max_arity () =
+  if !Clflags.native_code then 126 else max_int
```

Comment:
Leaving a note that this value is one fewer than might be expected, as a result of the hidden closure parameter, might be worthwhile.

Reply:
That was explained in the commit message.  But redundancy is good, see commit 25c85d65a .

Reply:
Pedantic eta note: If something makes sense both as a commit message and a code comment, I usually prefer the code comment. Both are visible at the same time to people reviewing the patch (so either location is fine), but code comments stay and remain easily visible in the future, while it is uncommon to blame into the past to read past commit messages.